apps/vanilla/src/pages/2/src/controllers/main-controller.ts

```diff
@@ -0,0 +1,81 @@
+import { render } from '../utils';
+import {
+	PlayerContainerView,
+	ScoresContainerView,
+	TitleView,
+	TotalScoresView,
+	RollButtonView,
+	PlayerView,
+} from '../views';
+import { PlayerGenerator, TurnGenerator, DiceGenerator } from '../generators';
+
+/** Controller.  */
+export class Controller {
+	/** Container of app. */
+	private app: HTMLElement;
+
+	/**  Container for players. */
+	private playersContainer: PlayerContainerView;
+
+	/** Button that generate game turn. */
+	private rollButton: RollButtonView;
+
+	/** Array of player generators. */
+	private readonly playersGenerators: PlayerGenerator[];
+
+	/** Array of player views. */
+	private readonly playersViews: PlayerView[];
```

Comment:
Don't forget adding `readonly` for arrays too.
```suggestion
	private readonly playersGenerators: readonly PlayerGenerator[];

	/** Array of player views. */
	private readonly playersViews: readonly PlayerView[];
```